src/parameter_validation.h

```diff
@@ -71,46 +71,26 @@ namespace Image_Function
             throw imageException( "Bad input parameters in image function" );
     }
 
-    template <typename TImage>
-    void ParameterValidation( const TImage & image1, const TImage & image2, const TImage & image3 )
+    template <typename TImage, typename... Args>
+    void ParameterValidation( const TImage & image1, const TImage & image2, Args... args )
     {
-        if( image1.empty() || image2.empty() || image3.empty() || !IsCorrectColorCount( image1 ) || !IsCorrectColorCount( image2 ) ||
-            !IsCorrectColorCount( image3 ) || image1.width() != image2.width() || image1.height() != image2.height() ||
-            image1.width() != image3.width() || image1.height() != image3.height() )
-            throw imageException( "Bad input parameters in image function" );
+        ParameterValidation( image1, image2 );
+        ParameterValidation( image2, args... );
     }
 
     template <typename TImage>
-    void ParameterValidation( const TImage & image, uint32_t startX, uint32_t startY, uint32_t width, uint32_t height )
+    void ParameterValidation( uint32_t width, uint32_t height, const TImage & image, uint32_t startX, uint32_t startY )
     {
         if( image.empty() || !IsCorrectColorCount( image ) || width == 0 || height == 0 || startX + width > image.width() || startY + height > image.height() ||
             startX + width < width || startY + height < height )
             throw imageException( "Bad input parameters in image function" );
     }
 
-    template <typename TImage>
-    void ParameterValidation( const TImage & image1, uint32_t startX1, uint32_t startY1,
-                              const TImage & image2, uint32_t startX2, uint32_t startY2,
-                              uint32_t width, uint32_t height )
-    {
-        if( image1.empty() || image2.empty() || !IsCorrectColorCount( image1 ) || !IsCorrectColorCount( image2 ) || width == 0 || height == 0 ||
-            startX1 + width > image1.width() || startY1 + height > image1.height() || startX1 + width < width || startY1 + height < height ||
-            startX2 + width > image2.width() || startY2 + height > image2.height() || startX2 + width < width || startY2 + height < height )
-            throw imageException( "Bad input parameters in image function" );
-    }
-
-    template <typename TImage>
-    void ParameterValidation( const TImage & image1, uint32_t startX1, uint32_t startY1,
-                              const TImage & image2, uint32_t startX2, uint32_t startY2,
-                              const TImage & image3, uint32_t startX3, uint32_t startY3,
-                              uint32_t width, uint32_t height )
+    template <typename TImage, typename... Args>
```

Comment:
Let me think please how to avoid such big change,,,

Reply:
Sure! I am happy to change this if there is a better way!

Reply:
Okay. I got an idea: we have an extra function which returns `std::pair<uint32_t, uint32_t>` value. Let's call it `std::pair<uint32_t, uint32_t> ExtractRoiSize( uint32_t width, uint32_t height )`.

Now we need to declare template function:
```cpp
template <typename TImage, typename... Args>
std::pair<uint32_t, uint32_t> ExtractRoiSize( const TImage & image1, uint32_t startX1, uint32_t startY1, Args... args )
{
    return ExtractRoiSize( args );
}
```

So then in your original function `ParameterValidation` you could get width and height

Reply:
It looks little ugly but we don't need to change the rest of library.

Reply:
I will give it a go, thanks!

Reply:
@ihhub Could you please have another look?